### PR TITLE
Refactor/error handling cr

### PR DESCRIPTION
**Pull request by Fordo29**
### 1. What changed?
We have added Error Handling and added additional testing to account for sad paths.

### 2. Why is this change necessary?
This was to complete our user testing and make a fluid experience for the user

### 3. What changed technically that may impact others?
I imported Error component into Pet and Home to be facilitate our query error handling.  We also added logic for when there are no pets to adopt. 

### 4. How do we test it?
Run All Cypress tests for this PR.  Error or sad paths were added.

### CODE REVIEW

File: cypress/integration/error_spec.js [hunk 1]
```diff
@@ -1,8 +1,48 @@
...
+      {
+        method: 'POST',
+        url: 'https://not-fur-gotten-be.herokuapp.com/graphql',
+        headers: {
+          'x-gql-operation-name': 'getAllPets',
+        },
+        times: 1
+      },
+      {
+        "errors": [
```
Comment (StevieAmb) [inline cypress/integration/error_spec.js:13]: This was a wonderful find for figuring out how to simulate an error in this GRAPHQL post request. Yay for the docs!

**Review by ectweitmann — APPROVED** [review]
WOW this great work Christine!! Way to go team!!!

File: src/Components/ApplicationsContainer.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ const ApplicationContainer = (apps) => {
       />
     )
   })
-  let display = allApplications.length ? allApplications : <p className='no-applications-section'>No applications quite yet.</p>
+  let display = allApplications.length ? allApplications : <p className='no-results'>No applications quite yet.</p>
```
Comment (StevieAmb) [inline src/Components/ApplicationsContainer.js:16]: Great job with this ternary!

File: src/Components/PetContainer.js [hunk 3]
```diff
@@ -4,19 +4,27 @@ import '../Styles/PetContainer.scss';
...
-        id={pet.id}
-        name={pet.name}
-        image={pet.image}
-      />
-    )
-  });
+  let petCards;
+  let emptyMessage;
+
+  if(!pets.length){
```
Comment (StevieAmb) [inline src/Components/PetContainer.js:10]: Nice work on this accounting for the sad paths if there are no pets for adoption.

**Review by StevieAmb — APPROVED** [review]

**Pull request merged**